Mathlib/SetTheory/Cardinal/Regular.lean

```diff
@@ -234,19 +234,43 @@ theorem deriv_lt_ord {f : Ordinal.{u} → Ordinal} {c} (hc : IsRegular c) (hc' :
 /-! ### Inaccessible cardinals -/
 
 /-- A cardinal is inaccessible if it is an uncountable regular strong limit cardinal. -/
-def IsInaccessible (c : Cardinal) :=
-  ℵ₀ < c ∧ IsRegular c ∧ IsStrongLimit c
+def IsInaccessible (c : Cardinal) : Prop :=
+  ℵ₀ < c ∧ c ≤ c.ord.cof ∧ ∀ x < c, 2 ^ x < c
```

Comment:
It's not clear to me that this is an improvement: I'd rather see definitions in mathlib use other, existing definitions in mathlib rather than reformulations of them which are later proven equivalent, unless there's some good reason not to do so.

Reply:
The idea here is to make the default constructor as minimal as possible. `IsRegular c` includes a `ℵ₀ ≤ c` assumption which is rendered redundant by `ℵ₀ < c`; `IsStrongLimit c` includes a `c ≠ 0` assumption which is the same. Otherwise, calling `constructor` on an `IsInaccessible` hypothesis becomes a bit of a noob trap, where you have to prove these three inequalities separately instead of just once.

I do think it's important to have the equivalence to the more mathematical definition, which I'm providing through `isInaccessible_def`.

Reply:
I agree with @vihdzp here not because it's a "noob trap" but because it's genuinely better to have a good default constructor that I can easily access with `\< \>` without having to use a custom constructor every time.

Further, if one wants to make the argument that with the old definition it was easy to deduce `IsRegular` from `h.2.1`, then my counter-argument would be that now she has provided the API for us to write `h.isRegular` instead which is arguably even better.